cmd/snap-update-ns/bootstrap.c

```diff
@@ -76,20 +83,10 @@ static int setns_into_snap(const char *snap_name)
 	return err;
 }
 
-// switch_to_privileged_user drops to the real user ID while retaining
+// switch_to_privileged_user drops to the specific user and group ID while retaining
 // CAP_SYS_ADMIN, for operations such as mount().
-static int switch_to_privileged_user()
+static int switch_to_specific_privileged_user(uid_t real_uid, gid_t real_gid)
 {
-	uid_t real_uid;
-	gid_t real_gid;
-
-	real_uid = getuid();
-	if (real_uid == 0) {
-		// We're running as root: no need to switch IDs
-		return 0;
-	}
```

Comment:
Why is this not needed any more? (I'm sorry, I'm finding the argument processing a little hard to follow in terms of what is absolutely allowable and what is implied (process_arguments() from bootstrap.c would ideally document all the valid invocations and error on invalid combinations as well as when the snap name is required and when it isn't). I understand that in bootstrap this is only called when uid != 0, but why not leave the `if (real_uid == 0) { return 0; }` check for future-proofing?